### PR TITLE
Adding chart for SignalFx agent

### PR DESCRIPTION
**Pull request by keitwb**
This is a new chart for our Kubernetes metric agent.  Our company is a new member of the CNCF, which I believe means we signed the Contributor agreement.

### CODE REVIEW

**Comment by unguiculus** [comment]
Thanks for the PR. Please make sure you read and understand our guidelines. You should base your chart on the initial setup you get with `helm create` using Helm 2.8+. And you'll have to get your CLA.

**Comment by jdumars** [comment]
Hi, please get your CLA signed, and then we'll ok this for testing.  Thanks!

**Comment by keitwb** [comment]
Ok, I'm working with my company's CNCF point person to link up my email to our corporate CLA signature so that I can sign this as an employee.

**Comment by keitwb** [comment]
Ok, I should be good to go with the CLA.

File: stable/signalfx-agent/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+keywords:
+- monitoring
+- alerting
+- metrics
+- stats
+home: https://signalfx.com
+sources:
+- https://github.com/signalfx/signalfx-agent
+maintainers:
+- name: Ben Keith
```
Comment (unguiculus) [inline stable/signalfx-agent/Chart.yaml]: Use Github username.

File: stable/signalfx-agent/templates/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{{- if .Values.rbac.create -}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "signalfx-agent.fullname" . }}
+  labels:
+    app: {{ template "signalfx-agent.name" . }}
+    {{ if ne .Release.Name "RELEASE-NAME" -}}
```
Comment (unguiculus) [inline stable/signalfx-agent/templates/clusterrole.yaml]: Remove these if statements. The labels should always be there.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please follow our best practices for RBAC.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/signalfx-agent/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    app: {{ template "signalfx-agent.name" . }}
+    version: {{ .Values.agentVersion }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    chart: {{ template "signalfx-agent.chart" . }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "signalfx-agent.name" . }}
+      {{ if ne .Release.Name "RELEASE-NAME" -}}
```
Comment (unguiculus) [inline stable/signalfx-agent/templates/daemonset.yaml]: Remove `if`

File: stable/signalfx-agent/templates/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: "100%"
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: {{ template "signalfx-agent.name" . }}
+        version: {{ .Values.agentVersion }}
+        {{ if ne .Release.Name "RELEASE-NAME" -}}
```
Comment (unguiculus) [inline stable/signalfx-agent/templates/daemonset.yaml]: Remove `if`

File: stable/signalfx-agent/templates/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/signalfx-agent/templates/daemonset.yaml]: `apps/v1beta2`

File: stable/signalfx-agent/templates/secrets.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: {{ template "signalfx-agent.fullname" . }}
+  labels:
+    app: {{ template "signalfx-agent.name" . }}
+    chart: {{ template "signalfx-agent.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+type: Opaque
+data:
+  access-token: {{ default "" .Values.signalFxAccessToken | b64enc | quote }}
```
Comment (unguiculus) [inline stable/signalfx-agent/templates/secrets.yaml]: Remove `default` function. It's unnecessary.

File: stable/signalfx-agent/values.yaml [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+agentVersion: 3.0.0
+
+# The access token for SignalFx.  REQUIRED
+signalFxAccessToken:
+
+# Image pull policy for the agent pod
+pullPolicy: IfNotPresent
+
+# The docker image to use for the agent
+imageRepository: quay.io/signalfx/signalfx-agent
```
Comment (unguiculus) [inline stable/signalfx-agent/values.yaml]: Check our review guidelines for how to configure the image.
https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/signalfx-agent/templates/daemonset.yaml (hunk 5 above)
Comment (keitwb) [reply stable/signalfx-agent/templates/daemonset.yaml]: Is it permissible to do a conditional here to support K8s 1.7 and older?  Or some other way to handle that?

**Comment by keitwb** [comment]
@unguiculus I made all of the changes you requested except for the one that I had a question on above.  What do you suggest about that?

File: stable/signalfx-agent/templates/daemonset.yaml (hunk 5 above)
Comment (timstoop) [reply stable/signalfx-agent/templates/daemonset.yaml]: As the Charts only support the last three stable versions, we now only support 1.8 to 1.10, right?

Comment (keitwb) [reply stable/signalfx-agent/templates/daemonset.yaml]: Ok, we'll just tell people the helm chart is only for newer versions, shouldn't be a problem.

**Comment by prydonius** [comment]
/ok-to-test

this lgtm, @unguiculus can you give it another quick look to see if your comments were addressed?

**Comment by prydonius** [comment]
@keitwb the tests are failing because the DaemonSet is not finding the ConfigMap which is not created if the required values are missing. To fix this, I recommend the following changes:

1. Use the same condition in the DaemonSet to skip creating it if required values are not provided
2. (Optionally) add a custom CI values.yaml with clusterName and the access token set to something - of course this may not be possible if there isn't a dummy access token that can be used in tests and would be safe to store in git.

**Comment by keitwb** [comment]
Ok, I did the first option since we don't have any kind of default values for the access token.  Thanks.

**Comment by prydonius** [comment]
@keitwb great, thanks!

/lgtm

**Pull request merged**